pages/best-practices/best-practices.mdx

```diff
@@ -0,0 +1,293 @@
+
+If an authorization concept can be expressed using relations, it should be.
+We provide caveats as an escape hatch; they should only be used for context that’s only available at request time, or else ABAC logic that cannot be expressed in terms of relationships.
+
+Some examples:
+
+- A banlist - this could be expressed as a list in caveat context, but it can also be expressed as a relation with negation.
+- A notion of public vs internal - boolean flags seem like an obvious caveat use case, but they can also be expressed using self relations.
+- Dynamic roles - these could be expressed as a list in caveats, and it’s not immediately obvious how to build them into a spicedb schema, but our [Google Cloud IAM example](https://authzed.com/blog/google-cloud-iam-modeling) shows how it’s possible.
+
+This is because caveats come with a performance penalty.
+A caveated relationship is both harder to cache and also slows down computation of the graph walk required to compute a permission.
+
+### Make Your Writes Idempotent
+
+Tags: **application**
+
+Relations in SpiceDB are binary (a relation is present or it's not), and `WriteRelationships` calls are atomic.
+As much as possible, we recommend that you use the [`TOUCH`](https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.RelationshipUpdate) semantic for your write calls, because it means that you can easily retry writes and recover from failures.
+
+If you’re concerned about sequencing your writes, or your writes have dependencies, we recommend using [preconditions](https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.Precondition).
+
+### Don’t truncate your tables when running Postgres
+
+Tags: **operations**
+
+If you truncate your Postgres table, your SpiceDB pods will become unresponsive until you run SpiceDB datastore repair.
+We recommend either dropping the tables entirely and recreating them with spicedb datastore migrate head or deleting the data using a `DeleteRelationships` call instead.
+
+To ensure that every request, whether cached or not, gets a consistent point-in-time view of the underlying data, SpiceDB uses Multi-Version Concurrency Control.
+Some datastores provide this natively; in others we’ve implemented it on top of the datastore.
+In Postgres, the implementation of MVCC depends on the internals of the transaction counter being stored as data in the tables, so if you truncate the relationships table you desync the transaction counter with the stored relationships.
+
+## Priority B Rules: Strongly Recommended
+
+### Understand your consistency needs
+
+Tags: **operations**
+
+While designing your authorization, it’s important to plan ahead for how quickly an update is guaranteed, while trading off performance via cache effectiveness.
+By default, SpiceDB sets the Quanitzation Interval to 5s; check operations are cached within this window.
+To change this value, set `--datastore-revision-quantization-interval` longer or shorter.
```

Comment:
I feel that this sentence doesn't help me decide _what_ value to use. A concrete example may help